src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -26,107 +34,172 @@
+
+    private static void validateParametersForUnauthorizedError(MultiMap queryParameters) {
+        final String account = queryParameters.get(ACCOUNT_PARAMETER);
+        if (StringUtils.isBlank(account)) {
+            throw new InvalidRequestException("'account' is required query parameter and can't be empty");
+        }
+    }
+
+    private Future<Boolean> isAccountEventEnabled(String accountId) {
+        return applicationSettings.getAccountById(accountId, timeoutFactory.create(DEFAULT_TIMEOUT))
+                .map(Account::getEventsEnabled)
+                .otherwise(throwable -> fallbackResult(accountId, throwable));
     }
 
-    private void validateEventRequestQueryParams(String format) {
-        if (format != null && !trackingPixels.containsKey(format)) {
-            throw new InvalidRequestException(String.format(
-                    "'format' query parameter can have one of the next values: %s", trackingPixels.keySet()));
+    private void handleEvent(AsyncResult<Boolean> isEnabledResult, boolean isAnalyticsEnabled, String format,
+                             RoutingContext context, HttpServerResponse response) {
+        if (isEnabledResult.result()) {
+            if (isAnalyticsEnabled) {
+                analyticsReporter.processEvent(makeNotificationEvent(context));
+                respondWithOkStatus(response, format);
+            }
+        } else {
+            respondWithUnauthorized(response, "Given 'accountId' is not supporting the event");
         }
     }
 
-    private void respondWithBadStatus(RoutingContext context, String message) {
-        context.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
-                .end(String.format("Request is invalid: %s", message));
+    private static NotificationEvent makeNotificationEvent(RoutingContext context) {
+        final MultiMap queryParameters = context.request().params();
+        final NotificationEvent.Type type = toNotificationType(queryParameters.get(TYPE_PARAMETER));
+        final String bidId = queryParameters.get(BID_ID_PARAMETER);
+        final String accountId = queryParameters.get(ACCOUNT_PARAMETER);
+        final HttpContext httpContext = HttpContext.from(context);
+        return NotificationEvent.of(type, bidId, accountId, httpContext);
     }
 
-    private void respondWithOkStatus(RoutingContext context, String format) {
-        if (format != null) {
-            final TrackingPixel trackingPixel = trackingPixels.get(format);
-            context.response()
-                    .putHeader(HttpHeaders.CONTENT_TYPE, trackingPixel.getContentType())
+    private static NotificationEvent.Type toNotificationType(String type) {
+        if (Objects.equals(WIN_TYPE, type)) {
+            return NotificationEvent.Type.win;
+        } else if (Objects.equals(IMP_TYPE, type)) {
+            return NotificationEvent.Type.imp;
+        }
+        throw new IllegalArgumentException(String.format("Type is undefined, %s", type));
```

Comment:
Who is catch this exception by? Pls add unit-test.

Reply:
This exception should never occurred bc we already check that in `validateParam()`.

Reply:
Changed to null.

Reply:
So, we have only two types and they are validated - do we need the second condition then?